crdsonnet/main.libsonnet

```diff
@@ -133,10 +134,16 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
                helpers.properties.withGroupVersionKind(gvk.group, gvk.version, gvk.kind)
              else {})
         );
-      processor.render(name, extendSchema)
-      + (if 'x-kubernetes-group-version-kind' in component
-         then processor.renderEngine.newFunction([name])
-         else processor.renderEngine.nilvalue),
+      if addNewFunction
+      then
+        // FIXME: this part doesn't work with AST render engine
+        processor.render(name, extendSchema)
+        + processor.renderEngine.toObject(
+          if 'x-kubernetes-group-version-kind' in component
+          then processor.renderEngine.newFunction([name])
+          else processor.renderEngine.nilvalue
+        )
+      else processor.render(name, extendSchema),
```

Comment:
Adding fields after the `render()` call keeps on being a nuisance and I might have made the wrong design choices by implementing them at this level, I think they'll need to end up in the render engine somewhere.

At this point I've wasted more time on this than I want to admit. This `addNewFunction` provides a stop-gap solution for the projects I'm currently working on, this should be sufficient for the peer review but should get fixed in a next iteration.